### PR TITLE
Add preliminary support for Matrix

### PR DESCRIPTION
**Pull request by mytbk**

### CODE REVIEW

**Comment by bigeagle** [comment]
wow, cooool!
Hmmm, have you tested it?

File: README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ big party again.
 
 - IRC
 - XMPP
+- [Matrix](https://matrix.org) (depends on [matrix-python-sdk](https://github.com/matrix-org/matrix-python-sdk))
```
Comment (bigeagle) [inline README.md]: Please add this dependency to `requirements.txt`

File: fishroom/matrix.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    def __init__(self, server, username, password, rooms):
+        client = MatrixClient(server)
+
+        try:
+            client.login_with_password(username, password)
+        except MatrixRequestError as e:
+            print(e)
+            if e.code == 403:
+                print("Bad username or password.")
```
Comment (bigeagle) [inline fishroom/matrix.py]: use `logger.info` instead :smiley_cat:

**Comment by mytbk** [comment]
The test() in matrix.py has not been tested yet. I've tested it with a Matrix room together with XMPP and IRC.

Multiple bindings has been tested with:

```
    'bindings': {
        "archlinux-cn": {
            "xmpp": 'archlinux@conference.cnjabber.net',
            "matrix": '#archlinux:matrixim.cc'
        },

        "linuxmasterrace": {
            "matrix": '#linuxmasterrace:matrix.org',
            "xmpp": 'linuxmasterrace@conference.cnjabber.net'
        }
    }
```


Only m.text msgtype in m.room.message event (https://matrix.org/docs/spec/client_server/r0.2.0.html#m-room-message-msgtypes) is supported.

Nick is not supported.

**Comment by bigeagle** [comment]
OK. I've commented on the code, you can fix the little problems that I've pointed out. I'll merge this as soon as you push the updates.

**Comment by mytbk** [comment]
I've added nick support in another commit.

bigeagle: I'm sorry, but I can't see where your comment is.

**Comment by bigeagle** [comment]
I'm sorry I forgot to submit my comments. :cry:

File: README.md (hunk 1 above)
Comment (mytbk) [reply README.md]: done

File: fishroom/matrix.py (hunk 2 above)
Comment (mytbk) [reply fishroom/matrix.py]: done

**Pull request merged**